### PR TITLE
[IMP] Improve assertRecordValues ouput for tests in common.py

### PR DESCRIPTION
**Pull request by smetl**
Display the differences in assertRecordValues instead of the whole inputs.

example: https://ibb.co/kMcztcG
The record is different at index 3. There is a new not-expected record at index 4.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/tests/common.py [hunk 1]
```diff
@@ -309,64 +309,93 @@ def assertRecordValues(self, records, expected_values):
...
+                        if record_currency.compare_amounts(candidate[field_name], record_value):
+                            diff[field_name] = (record_value, record_currency.round(candidate[field_name]))
+                    elif candidate[field_name] != record_value:
+                        diff[field_name] = (record_value, candidate[field_name])
                 elif field_type in ('one2many', 'many2many'):
                     # Compare x2many relational fields.
                     # Empty comparison must be an empty list to be True.
-                    if set(record_value.ids) != set(candidate_value):
-                        return False
+                    if field_name not in candidate:
```
Comment (oco-odoo) [inline odoo/tests/common.py]: This seems to start each of the sub-conditions; couldn't it be put just once, before "if field_type == 'monetary'" ?

File: odoo/tests/common.py [hunk 2]
```diff
@@ -309,64 +309,93 @@ def assertRecordValues(self, records, expected_values):
...
+        if not different_values:
+            return
 
-        def _format_message(records, expected_values):
-            ''' Return a formatted representation of records/expected_values. '''
-            all_records_values = [{key: _repr_field_value(record, key) for key in expected_values[0]} for record in records]
-            msg1 = '\n'.join(pprint.pformat(dic) for dic in all_records_values)
-            msg2 = '\n'.join(pprint.pformat(dic) for dic in expected_values)
-            return 'Current values:\n\n%s\n\nExpected values:\n\n%s' % (msg1, msg2)
+        errors = ['The records and expected_values does not match.']
```
Comment (oco-odoo) [inline odoo/tests/common.py]: do not match*

File: odoo/tests/common.py (hunk 1 above)
Comment (smetl) [reply odoo/tests/common.py]: Not really because the difference to add is different depending the field type :/

**Comment by oco-odoo** [comment]
@robodoo r+

**Comment by smetl** [comment]
@robodoo retry

**Pull request closed without merging**